### PR TITLE
Remove PTofRE implementations from source. Tests pass on my p9 darwin…

### PR DESCRIPTION
**Pull request by dholladay00**
… node.

## PR Summary
As discussed with @Yurlungur, this function was the sole reason for very excessive GPU compile times. The singular place it was used was replaced with an equivalent functionality. It does not appear to be used anywhere else. 

This PR removes `PTofRE` function. I'd like to hear from folks downstream, @jdolence @jhp-lanl do you foresee issues with the removal of this function to downstream use cases and/or future use cases (PT table equivalent for example @jhp-lanl?)

## PR Checklist

- [x] Adds a test for any bugs fixed. Adds tests for new features.
- [x] Format your changes by using the `make format` command after configuring with `cmake`.
- [x] Document any new features, update documentation for changes made.
- [x] Make sure the copyright notice on any files you modified is up to date.
- [x] After creating a pull request, note it in the CHANGELOG.md file
- [x] If preparing for a new release, update the version in cmake.

### CODE REVIEW

**Comment by Yurlungur** [comment]
To add some context, this function was originally written as a convenience for the flag-type PTE solver. But the current solvers don't use this function, and the same functionality is available as a collection of calls, or through FillEos. 

While I believe we may need functions that compute quantities given Pressure as an input, this function doesn't provide it. So I see this as just removing code that (a) we don't need and shouldn't continue to maintain and (b) apparently slows compile times way down for some reason.

**Review by Yurlungur — APPROVED** [review]
The list of files changed doesn't seem like the entire list of Eoses and modifiers. Let's just make sure we got them all.

**Comment by dholladay00** [comment]
> The list of files changed doesn't seem like the entire list of Eoses and modifiers. Let's just make sure we got them all.

I found things to remove via grep, so I think things that didn't include probably relied on the base implementation.

**Comment by jhp-lanl** [comment]
Obviously not a blocker for this MR: I'd like to understand _why_ this was so bad at some point though so that we know what we can't feasibly do in the CRTP base class. Like is it a problem to call multiple class member functions there? Or is it a combination of that with the parallel for in the vector version? I suppose we'll find out whenever we try something that drastically increases compilation times 🙃

File: doc/sphinx/src/using-eos.rst [hunk 1]
```diff
@@ -549,10 +549,3 @@ returns the density at which pressure is minimized for a given
 temperature. This is again useful for root finds.
 
 Finally the method
```
Comment (jhp-lanl) [inline doc/sphinx/src/using-eos.rst:551]:
```suggestion

```

**Review by jhp-lanl — APPROVED** [review]
I think you left a dangling remnant of the documentation. Otherwise looks good

**Comment by Yurlungur** [comment]
@dholladay00 is this ready to go? I re-triggered the re-git pipelline

**Comment by Yurlungur** [comment]
Tests passing. Pulling the trigger.

**Pull request merged**